blockchains/erc20/lib/fetch_events.js

```diff
@@ -228,15 +204,14 @@ async function getRawEvents(web3, fromBlock, toBlock, contractAddress) {
   return await web3.eth.getPastLogs(queryObject);
 }
 
-async function decodeEvents(web3, events, decodeFunctions=decodeFunctionsMap) {
-  const blockTimestamps = {};
```

Comment:
Previously that object was the cache. However that was not optimal for the case where we extract per contract, because we arrive in this method for each contract. This invalidates the cache when it can be still used.